### PR TITLE
switch kube subnet manager to PATCH

### PR DESCRIPTION
**Pull request by mikedanese**
This works too.

@aaronlevy

### CODE REVIEW

**Comment by philips** [comment]
This LGTM but I haven't tested it.

File: subnet/kube/kube.go [hunk 1]
```diff
@@ -231,7 +233,31 @@ func (ksm *kubeSubnetManager) AcquireLease(ctx context.Context, network string,
...
+		if err != nil {
+			return nil, err
+		}
+
+		patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldData, newData, v1.Node{})
+		if err != nil {
+			return nil, fmt.Errorf("failed to create patch for node %q: %v", ksm.nodeName, err)
+		}
+
+		_, err = ksm.client.Core().Nodes().Patch(ksm.nodeName, api.StrategicMergePatchType, patchBytes, "status")
```
Comment (aaronlevy) [inline subnet/kube/kube.go]: What does providing the "status" subresource do/mean?

Comment (mikedanese) [inline subnet/kube/kube.go]: @aaronlevy it's a subresource under node that only allows updates to .status fields and annotations. It's less privileged then the node resource and can be acl'd seperately. This code was written pre-rbac and I'm trying to reduce the privilege required by kube-flannel.

https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/node/strategy.go#L122 status strategy zeros out changes to spec.

Comment (mikedanese) [inline subnet/kube/kube.go]: Also see changes to rbac file.

Comment (aaronlevy) [inline subnet/kube/kube.go]: ah k. sgtm

**Comment by aaronlevy** [comment]
@diegs and/or I will try and start testing this and https://github.com/coreos/flannel/pull/677 today.

**Comment by mikedanese** [comment]
I tested this by hand on friday. We should probably merge this, cut a release then consider merging #677.

**Comment by aaronlevy** [comment]
The changes themselves LGTM to me then.

**Comment by mikedanese** [comment]
(more hand testing is always appreciated)

**Comment by diegs** [comment]
Also hand-tested this one: launched a node using `--register-with-taints=key=value:PreferNoSchedule`. Verified that after flannel configured the node the taint was still present. Did a bunch of host-pod and pod-pod network checks and all looked good.

**Pull request merged**